packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1300,7 +1459,32 @@ function validateTemplate(
     }
 }
 
-function validateChildren(ctx: ParserCtx, element?: BaseElement): void {
+function validateChildren(ctx: ParserCtx, element?: BaseElement, directive?: ParentNode): void {
+    if (directive) {
+        // Find a scoped slot fragment node if it exists
+        const slotFragment = ctx.findAncestor(
+            ast.isScopedSlotFragment,
+            ({ current }) => current && ast.isComponent,
+            directive
+        );
+
+        if (!isNull(slotFragment)) {
+            slotFragment.children.forEach((child) => {
+                // Error in compiler logic
+                if (ast.isElementDirective(child)) {
+                    throw new Error(
```

Comment:
Could you add a `/* istanbul ignore if */` here?  

We've been adding it in the template compiler for internal compiler errors so that the coverage report doesn't pick it up.